File: pteid-mw-pt/_src/eidmw/eidlibNodeJS_Wrapper/examples/simple_example.js
#!/usr/bin/env node

var pteid = require('../build/Release/pteid');

function init() {
  console.log('CC sdk example, starting...');
  pteid.PTEID_ReaderSet.initSDK();
}

function release() {
  console.log('CC sdk example, releasing resources...');
  pteid.PTEID_ReaderSet.releaseSDK();
}

function start() {
  var readers,
      count,
      device_names,
      idx_reader_name,
      ctx_with_card,
      reader_ctx,
      card,
      card_id,
      idx;

  readers = pteid.PTEID_ReaderSet.instance();

  count = readers.readerCount();
  console.log('Number of connected card readers to the system:', count);

  if (count) {
    device_names = [];
    for (idx = 0; idx < count; idx++) {
      console.log('Obtaining card name for index:', idx);

      idx_reader_name = readers.getReaderName(idx);
      device_names.push(idx_reader_name);

      console.log('Connected device:', idx_reader_name);
    }

    ctx_with_card = null;
    for (idx = 0; idx < device_names.length; idx++) {
      reader_ctx = readers.getReaderByName(device_names[idx]);

      if (reader_ctx.isCardPresent()) {
        console.log('The reader <', device_names[idx], '> has a card in it.');

        ctx_with_card = reader_ctx;
        break;
      }
    }

    if (ctx_with_card !== null) {
      card = ctx_with_card.getEIDCard();
      card_id = card.getID();

      dump_card_info(card_id);
    } else {
      console.log("There's no card present in the reader, terminating...");
    }
  }
}

function dump_card_info(id) {
  console.log('Card ID:', id);
  console.log("-------------------------------------------------------");
  console.log("deliveryEntity             ", id.getIssuingEntity());
  console.log("country                    ", id.getCountry());
  console.log("documentType               ", id.getDocumentType());
  console.log("cardNumber                 ", id.getDocumentNumber());
  console.log("cardNumberPAN              ", id.getDocumentPAN());
  console.log("cardVersion                ", id.getDocumentVersion());
  console.log("deliveryDate               ", id.getValidityBeginDate());
  console.log("locale                     ", id.getLocalofRequest());
  console.log("validityDate               ", id.getValidityEndDate());
  console.log("name                       ", id.getSurname());
  console.log("firstname                  ", id.getGivenName());
  console.log("sex                        ", id.getGender());
  console.log("nationality                ", id.getNationality());
  console.log("birthDate                  ", id.getDateOfBirth());
  console.log("height                     ", id.getHeight());
  console.log("numBI                      ", id.getCivilianIdNumber());
  console.log("nameFather                 ", id.getSurnameFather());
  console.log("firstnameFather            ", id.getGivenNameFather());
  console.log("nameMother                 ", id.getSurnameMother());
  console.log("firstnameMother            ", id.getGivenNameMother());
  console.log("numNIF                     ", id.getTaxNo());
  console.log("numSS                      ", id.getSocialSecurityNumber());
  console.log("numSNS                     ", id.getHealthNumber());
  console.log("Accidental indications     ", id.getAccidentalIndications());
  console.log("mrz1                       ", id.getMRZ1());
  console.log("mrz2                       ", id.getMRZ2());
  console.log("mrz3                       ", id.getMRZ3());
  console.log("-------------------------------------------------------");
}

init();
start();
release();

File: pteid-mw-pt/_src/eidmw-branch/eidlibPython_Wrapper/examples/simple_example.py
#!/usr/bin/env python3

import sys
sys.path.append("../GeneratedModule")
import pteid

def init():
  # First of all you always have to initialize the sdk.
  print("CC sdk example, starting...")
  pteid.PTEID_ReaderSet.initSDK()

def release():
  print("CC sdk example, releasing resources...")
  # At the end, always release the sdk
  pteid.PTEID_ReaderSet.releaseSDK()

def start():
  # Obtain the instance of readerSet
  readers = pteid.PTEID_ReaderSet.instance()

  count = readers.readerCount()
  print("Number of connected card readers to the system:", count)

  # Check if we have connected smart card readers
  if count:
    devices_names = []

    # Obtain the names of all connected readers
    for idx in range(count):
      print("Obtaining card name for index:", idx)

      name = readers.getReaderName(idx)
      devices_names.append(name)

      print("Connected device:", name)

    ctx_with_card = None
    # try to find a smart card in one of the detected readers
    for name in devices_names:
      reader_ctx = readers.getReaderByName(name)
      
      if reader_ctx.isCardPresent():
        print("The reader <", name, "> has a card in it.")
       
        ctx_with_card = reader_ctx
        # found a card connected, don't look around in other readers
        break

    # Now check if we have a card to access we can keep going
    if ctx_with_card:
      card = ctx_with_card.getEIDCard()
      card_id = card.getID()

      dump_card_info(card_id)
    else:
      print("There's no card present in the reader, terminating...")

def dump_card_info(id):
  print("Card ID: ", id)
  print("-------------------------------------------------------") 
  print("deliveryEntity             ", id.getIssuingEntity())
  print("country                    ", id.getCountry())
  print("documentType               ", id.getDocumentType())
  print("cardNumber                 ", id.getDocumentNumber())
  print("cardNumberPAN              ", id.getDocumentPAN())
  print("cardVersion                ", id.getDocumentVersion())
  print("deliveryDate               ", id.getValidityBeginDate())
  print("locale                     ", id.getLocalofRequest())
  print("validityDate               ", id.getValidityEndDate())
  print("name                       ", id.getSurname())
  print("firstname                  ", id.getGivenName())
  print("sex                        ", id.getGender())
  print("nationality                ", id.getNationality())
  print("birthDate                  ", id.getDateOfBirth())
  print("height                     ", id.getHeight())
  print("numBI                      ", id.getCivilianIdNumber())
  print("nameFather                 ", id.getSurnameFather())
  print("firstnameFather            ", id.getGivenNameFather())
  print("nameMother                 ", id.getSurnameMother())
  print("firstnameMother            ", id.getGivenNameMother())
  print("numNIF                     ", id.getTaxNo())
  print("numSS                      ", id.getSocialSecurityNumber())
  print("numSNS                     ", id.getHealthNumber())
  print("Accidental indications     ", id.getAccidentalIndications())
  print("mrz1                       ", id.getMRZ1())
  print("mrz2                       ", id.getMRZ2())
  print("mrz3                       ", id.getMRZ3())
  print("-------------------------------------------------------") 

if __name__ == "__main__":
  init()
  start()
  release()
